### ISSUE TITLE
Dart format not working on list of records in Flutter project

### ISSUE THREAD

**Issue by JanDrabik** [dart-lang/dart_style#1231]
When formating code that has declaration such as 
```dart
class _FiltersPageState extends State<FiltersPage> {
  late TextEditingController controller;
  List<(String, String)> data = [('Temperature', '3758'),('Temperature range', '2930'),];

  @override
  void initState() {
    controller = TextEditingController();
    super.initState();
  }

  @override
  void dispose() {
    controller.dispose();
    super.dispose();
  }

// [...] build method
```
it is transformed into
```dart
  List<(String, String)

  >

  data

  =

  [

  (

  '

  Temperature

  '

  ,

  '

  3758

  '

  )

  ,

  ('Temperature range', '2930'),];
```

```
❯ flutter doctor
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.10.3, on macOS 13.3.1 22E261 darwin-arm64, locale en-PL)
[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.2)
[✓] Xcode - develop for iOS and macOS (Xcode 14.3)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2022.2)
[✓] Android Studio (version 2022.2)
[✓] VS Code (version 1.77.0)
[✓] Connected device (3 available)
[✓] Network resources

• No issues found!
```
My flutter project uses:
```dart
environment:
  sdk: '>=3.0.2 <4.0.0'
  flutter: ">=3.7.7"
```

**Comment by munificent** [comment]
Weird output like that where each token ends up on its own line isn't coming from `dart format`. Instead, it's a quirk of some IDE integration's built-in formatting support when it gets confused. Are you using IntelliJ? I think that's where I've seen output like this. If I run your example through `dart format` on the command line, I get:

```dart
class _FiltersPageState extends State<FiltersPage> {
  late TextEditingController controller;
  List<(String, String)> data = [
    ('Temperature', '3758'),
    ('Temperature range', '2930'),
  ];

  @override
  void initState() {
    controller = TextEditingController();
    super.initState();
  }

  @override
  void dispose() {
    controller.dispose();
    super.dispose();
  }

  // [...] build method
}
```

Closing this because there's nothing in dart_style to fix here.
